docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+
+#### KeyGen starts in Concrete
+
+First, we generate the Concrete keyset and then serialize the shared secret key that will be used to encrypt the inputs. In our case, this shared secret key is the same for all inputs and outputs.
+
+```python
+# generate all keys from scratch (not using initial secret keys)
+circuit.keygen()
+# since both inputs have the same type, they will use the same secret key, thus we serialize it once
+secret_key: bytes = tfhers_bridge.serialize_input_secret_key(input_idx=0)
+# we write it to a file to be used by TFHE-rs
+with open("secret_key_from_concrete", "wb") as f:
+    f.write(secret_key)
+```
+
+Next, we generate client and server keys in TFHE-rs using the shared secret key from Concrete. We will cover serialization in a [later section](#serialization-of-ciphertexts-and-keys), so there's no need to worry about how we loaded the secret key.
+
+```rust
+use tfhe::core_crypto::prelude::LweSecretKey;
+use tfhe::ClientKey;
+
+/// ...
+
+let lwe_sk: LweSecretKey<Vec<u64>> = load_lwe_sk("secret_key_from_concrete");
+let shortint_key =
+    tfhe::shortint::ClientKey::try_from_lwe_encryption_key(
+        lwe_sk,
+        // Concrete uses this parameters to define the TFHE-rs ciphertext type
+        tfhe::shortint::prelude::PARAM_MESSAGE_2_CARRY_3_KS_PBS
+    ).unwrap();
+let client_key = ClientKey::from_raw_parts(shortint_key.into(), None, None);
+let server_key = client_key.generate_server_key();
+
+```
+
+#### KeyGen starts in TFHE-rs
+
+First, we generate the TFHE-rs keyset and then serialize the shared secret key that will be used to encrypt the inputs
+
+```rust
+
+use tfhe::{prelude::*, ConfigBuilder};
+use tfhe::generate_keys;
+
+/// ...
+
+// Concrete uses this parameters to define the TFHE-rs ciphertext type
+let config = ConfigBuilder::with_custom_parameters(
+        tfhe::shortint::prelude::PARAM_MESSAGE_2_CARRY_3_KS_PBS,
+    )
+.build();
+
+let (client_key, server_key) = generate_keys(config);
+let (integer_ck, _, _) = client_key.clone().into_raw_parts();
+let shortint_ck = integer_ck.into_raw_parts();
+let (glwe_secret_key, _, _) = shortint_ck.into_raw_parts();
+let lwe_secret_key = glwe_secret_key.into_lwe_secret_key();
+
+save_lwe_sk(lwe_secret_key, "secret_key_from_tfhers");
+
+```
+
+Next, we generate a Concrete keyset using the shared secret key from TFHE-rs.
+
+```python
+# this was generated from TFHE-rs
+with open("secret_key_from_tfhers", "rb") as f:
+    sk_buff = f.read()
+# maps input indices to their secret key
+input_idx_to_key = {0: sk_buff, 1: sk_buff}
+# we do a Concrete keygen starting with an initial set of secret keys
+tfhers_bridge.keygen_with_initial_keys(input_idx_to_key_buffer=input_idx_to_key)
+```
+
+### Using Ciphertexts
+
+We should have everything now to encrypt, compute and decrypt on both sides. We will now do an encryption and computation in TFHE-rs, move to Concrete, run our circuit, and come back to TFHE-rs for decryption.
```

Comment:
could you precise if it's for 1.1, 1.2 or both? in general we should try to clarify what's common and what's specific